### PR TITLE
use pathlib in tools/gh_api.py

### PR DESCRIPTION
**Pull request by PhanatosZou**
This is related to #12515 
This is my first contribution here so please guide me if I'm doing anything wrong.
@Carreau, please review this PR.
Thank you.

### CODE REVIEW

**Comment by PhanatosZou** [comment]
Kindly ask if anyone know what's wrong here? Seems like it fails the test with Python 3.8. Any advice on improvement will be appreciated. Thanks!

**Comment by MrMino** [comment]
The CI failure looks unrelated. The failure is here: https://travis-ci.com/github/ipython/ipython/jobs/479778266#L773

File: tools/gh_api.py [hunk 1]
```diff
@@ -276,8 +276,7 @@ def post_download(project, filename, name=None, description=""):
     """Upload a file to the GitHub downloads area"""
     if name is None:
         name = Path(filename).name
-    with open(filename, 'rb') as f:
-        filedata = f.read()
+    filedata = Path(filename).read_bytes()
```
Comment (MrMino) [inline tools/gh_api.py:279]: You could improve this method further by making sure it creates `Path` only once - currently it does twice.

**Review by MrMino — CHANGES_REQUESTED** [review]

**Comment by Carreau** [comment]
Thanks for your patience and sending this; I was not able to keep up with everything and to keep the number of PRs small I'm going to close this one; If you believe this was a mistake, feel free to comment or re-send it; 

I can't guarantee I'll get to it as I do not have the bandwidth.

**Pull request closed without merging**